### PR TITLE
Social Authentication added

### PR DESCRIPTION
**Pull request by Ansh101112**
## Fixes Issue
#1825 

## Summary by CodeRabbit

- **New Features**
  - Added Google OAuth2 authentication for streamlined login and signup.
  - Introduced session management for enhanced user experience.
  - New route for updating user details.
  - New "AdditionalDetails" component for extra user data submission.

- **Improvements**
  - Updated login and signup flows to include social login options.
  - Improved handling of email and username in the dashboard.

- **Bug Fixes**
  - Fixed typo in the `handleLogout` function and enhanced error handling.

- **User Interface**
  - Updated sidebar styling and added hover effects to icons.
  - Improved button functionality and formatting in signup and login forms.

### CODE REVIEW

File: backend/routes/userRoutes.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+      user.phone = phone;
+      await user.save();
+      res.status(200).send("Details updated successfully");
+    } else {
+      res.status(404).send("User not found");
+    }
+  } catch (error) {
+    res.status(500).send(error.message);
+  }
+});
```
[the remark this answers is not in the document]
Comment (codervivek5) [reply backend/routes/userRoutes.js:20]: check this code

**Comment by codervivek5** [comment]
Kindly solve merge conflict

**Comment by Ansh101112** [comment]
> Kindly solve merge conflict

Done sir kindly see.

File: backend/index.js [hunk 2]
```diff
@@ -1,23 +1,29 @@
...
 dotenv.config();
 
 const app = express();
 
-// Middleware
 app.use(express.json());
 app.use(cors());
-// Routes
-app.use("/api", routes);
-app.use("/api/admin", adminRouter);
```
Comment (codervivek5) [inline backend/index.js:18]: ise hta diya kya ....admin routes

File: backend/models/User.js [hunk 3]
```diff
@@ -10,29 +12,34 @@ const userSchema = new Schema({
...
+    required: false, // Changed to false
     unique: true,
     trim: true,
     lowercase: true,
     match: [/^\S+@\S+\.\S+$/, "Please enter a valid email address"],
   },
   password: {
     type: String,
-    required: true,
+    required: false, // Changed to false
```
Comment (codervivek5) [inline backend/models/User.js:23]: ise false ku kr diya ye to required hi sahi rhega n ??

File: backend/index.js (hunk 2 above)
Comment (Ansh101112) [reply backend/index.js:18]: Yes sir kyuki abhi change krna hoga 3 admin ke liye different api bnegi

Comment (Ansh101112) [reply backend/index.js:18]: > ise hta diya kya ....admin routes

And in next pr I will try to add all admin routes accordingly to need

File: backend/models/User.js (hunk 3 above)
Comment (Ansh101112) [reply backend/models/User.js:23]: Yes sir but Google authentication ke time pe error de raha tha tbhi try Kiya false krke tb chal raha tha

Comment (codervivek5) [reply backend/models/User.js:23]: but ye to fix se ek bug bn gya then iska kya ?

Comment (Ansh101112) [reply backend/models/User.js:23]: Will change according to the new model as discussed in meeting now.

**Review by codervivek5 — APPROVED** [review]

**Pull request merged**